drivers/gpio/gpio_silabs.c

```diff
@@ -281,9 +284,7 @@ static int gpio_silabs_pin_interrupt_configure(const struct device *dev, gpio_pi
 		}
 
 		common->irq_pin_map[int_no] = pin;
```

Comment:
When I read the code, it is not super clear there is no risk to overwrite an existing mapping. I believe we could check `common->irq_pin_map[int_no] == 0xFF` before to write it.

 I believe it would be possible to retrieve the pin associated to an IRQ using registers  `EXTIPINSEL[LH]`. However, this is not trivial and there is no function in the `sl_hal_gpio` to do that (personally I would add a TODO).